src/FA/FiniteAutomata.java

```diff
@@ -0,0 +1,137 @@
+package FA;
+
+import domain.Pair;
+
+import java.io.*;
+import java.util.*;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.stream.Collectors;
+
+public class FiniteAutomata {
+    private List<String> Q;
+    private List<String> E;
+    private HashMap<Pair<String, String>, List<String>> D;
+    private String q0;
+    private List<String> F;
+
+    public List<String> getQ() {
+        return Q;
+    }
+
+    public List<String> getE() {
+        return E;
+    }
+
+    public HashMap<Pair<String, String>, List<String>> getD() {
+        return D;
+    }
+
+    public String getQ0() {
+        return q0;
+    }
+
+    public List<String> getF() {
+        return F;
+    }
+
+    public FiniteAutomata() {
+        Q = new ArrayList<>();
+        E = new ArrayList<>();
+        D = new HashMap<>();
+        F = new ArrayList<>();
+    }
+
+    public List<String> lineSplitter(String line) {
+        return Arrays.stream(line.strip().split(" ")).skip(2L).collect(Collectors.toList());
+    }
+
+    public List<Pair<String, String>> getAllTransition(String nextState, String how) {
+        Set<Pair<String, String>> allStateTransitions = D.keySet();
+        return allStateTransitions.stream().filter(transition -> transition.getKey().equals(nextState) && D.get(transition).contains(how)).collect(Collectors.toList());
+    }
+
+    public boolean acceptsSequence(String check) {
+        if (isDFA()) {
+            List<Pair<String, String>> nextMoves;
+            boolean found = false;
+            nextMoves = getAllTransition(q0, check.substring(0, 1));
+            Queue<Pair<String, String>> queue = new LinkedList<>(nextMoves);
```

Comment:
A queue gets a linked list? Very weird

Reply:
this is the java way, linkedlist implements the queue interface